### PR TITLE
Dilpreet | Stage | Fixed Prod Bugs

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/actions/login.action.ts [hunk 1]
```diff
@@ -342,10 +342,9 @@ export class LoginActions {
         .pipe(
             ofType(LoginActions.LogOut),
             map((action: CustomActions) => {
-                const countryRegion = localStorage.getItem('Country-Region');
-                const region = COUNTRY_REGION_MAP[countryRegion] || null;
                 if (PRODUCTION_ENV && !isElectron) {
-                    window.location.href = region === "gl" ? "https://giddh.com/login" : `https://giddh.com/${region}/login`;
+                    let giddhRegion = this._generalService.getGiddhRegionUrl();
+                    window.location.href = giddhRegion;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/actions/login.action.ts]:
```suggestion
                    window.location.href = this._generalService.getGiddhRegionUrl();
```

File: apps/web-giddh/src/app/app.component.ts [hunk 2]
```diff
@@ -82,7 +82,10 @@ export class AppComponent implements AfterViewInit, OnInit, OnDestroy {
         if (!(this._generalService.user && this._generalService.sessionId)) {
             if (!window.location.href.includes('login') && !window.location.href.includes('token-verify') && !window.location.href.includes('download') && !window.location.href.includes('verify-subscription-ownership') && !window.location.href.includes('dns')) {
                 if (PRODUCTION_ENV && !isElectron) {
-                    window.location.href = 'https://giddh.com/login/';
+                    const countryRegion = localStorage.getItem('Country-Region');
+                    const region = COUNTRY_REGION_MAP[countryRegion] || null;
+                    let giddhRegion = region === 'gl' ? 'https://giddh.com/login/' : `https://giddh.com/${region}/login/`;
+                    window.location.href = giddhRegion;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/app.component.ts]: use. common logic using common funciton

File: apps/web-giddh/src/app/login/login.component.ts [hunk 3]
```diff
@@ -259,7 +261,8 @@ export class LoginComponent implements OnInit, OnDestroy {
         });
 
         if (PRODUCTION_ENV && !isElectron) {
-            window.location.href = 'https://giddh.com/login';
+            let giddhRegion = this.generalService.getGiddhRegionUrl();
+            window.location.href = giddhRegion;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/login/login.component.ts]:
```suggestion
            window.location.href = this.generalService.getGiddhRegionUrl();
```

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 4]
```diff
@@ -797,7 +797,8 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
                 if (isElectron) {
                     this.router.navigate(['/login']);
                 } else {
-                    window.location.href = (environment.production) ? `https://giddh.com/login` : `https://test.giddh.com/login`;
+                    let giddhRegion = this.generalService.getGiddhRegionUrl();
+                    window.location.href = (environment.production) ? giddhRegion : `https://test.giddh.com/login`;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/header/header.component.ts]: Use common logic in overall giddh

File: apps/web-giddh/src/app/actions/login.action.ts (hunk 1 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/actions/login.action.ts]: DONE

File: apps/web-giddh/src/app/app.component.ts (hunk 2 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/app.component.ts]: Done

**Review by dvCodeWorld — DISMISSED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**